src/plugins/ab-testing.js

```diff
@@ -0,0 +1,59 @@
+import experiments from '../experiments';
+
+export default function(ctx, inject) {
+  // is there an experiment?
+  if (experiments.length === 0) {
+    inject('experiments', () => {
+      return false;
+    });
+    return false;
+  }
+  // check consent
+  const consentCookie = ctx.$cookies.get('klaro');
+  if (!consentCookie?.matomo) {
+    inject('experiments', () => {
+      return false;
+    });
+    return false;
+  }
```

Comment:
Why inject useless functions? 
```
  if (experiments.length === 0 || !ctx.$cookies.get('klaro')?.matomo) {
    return;
  }
```
In layout/default.vue we can then use `this.$experiments?.()` to first check if the function exists (is injected) and then call it.